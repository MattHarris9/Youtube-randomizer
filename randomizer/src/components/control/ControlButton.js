import React from 'react';
import PropTypes from 'prop-types';
import {
  BUTTON_HOVER,
  BUTTON_COLOR,
  BUTTON_ACTIVE,
  BUTTON_ACTIVE_BG,
} from '../../constants/colors';
import IconButton from '@material-ui/core/IconButton';
import Grid from '@material-ui/core/Grid';
import { styled } from '@material-ui/core';
import Tooltip from '@material-ui/core/Tooltip';

const StyledIconButton = styled(IconButton)({
  '& > span > svg': {
    color: BUTTON_COLOR,
  },
  '&:hover': {
    backgroundColor: BUTTON_HOVER,
    color: '#fff !important',
  },
  '&:hover > span > svg': {
    color: '#fff',
  },
});

const StyledSelectedIconButton = styled(IconButton)({
  '&': {
    backgroundColor: BUTTON_ACTIVE_BG,
  },
  '& > span > svg': {
    color: BUTTON_ACTIVE,
  },
  '&:hover > span > svg': {
    color: '#fff',
  },
  '&:hover': {
    backgroundColor: BUTTON_HOVER,
    color: '#fff !important',
  },
});

class ControlButton extends React.Component {
  static propTypes = {
    tooltip: PropTypes.string,
    icon: PropTypes.object,
    click: PropTypes.func,
    active: PropTypes.bool,
  };

  render() {
    const { tooltip, click, icon, active = false } = this.props;

    return (
      <Grid item>
        <Tooltip title={tooltip || ''}>
          {active ? (
            <StyledSelectedIconButton onClick={click}>
              {!!icon && icon}
            </StyledSelectedIconButton>
          ) : (
            <StyledIconButton onClick={click}>{!!icon && icon}</StyledIconButton>
          )}
        </Tooltip>
      </Grid>
    );
  }
}

export default ControlButton;
