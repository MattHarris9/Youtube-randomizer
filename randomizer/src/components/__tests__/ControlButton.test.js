import React from 'react';
import renderer from 'react-test-renderer';
import ControlButton from '../control/ControlButton';
import ShuffleIcon from '@material-ui/icons/Shuffle';

const mockFn = jest.fn();

describe('Control Button', () => {
  test('renders empty control button correctly', () => {
    const tree = renderer.create(<ControlButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders full control button correctly', () => {
    const props = {
      tooltip: 'Tooltip',
      click: mockFn,
      icon: <ShuffleIcon />,
      active: false,
    };
    const tree = renderer.create(<ControlButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
