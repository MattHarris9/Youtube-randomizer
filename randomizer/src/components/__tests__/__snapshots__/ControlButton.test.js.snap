// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control Button renders empty control button correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
>
  <button
    aria-describedby={null}
    className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    title=""
    type="button"
  >
    <span
      className="MuiIconButton-label"
    />
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Control Button renders full control button correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
>
  <button
    aria-describedby={null}
    className="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-112"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    title="Tooltip"
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M10.59 9.17L5.41 4 4 5.41l5.17 5.17 1.42-1.41zM14.5 4l2.04 2.04L4 18.59 5.41 20 17.96 7.46 20 9.5V4h-5.5zm.33 9.41l-1.41 1.41 3.13 3.13L14.5 20H20v-5.5l-2.04 2.04-3.13-3.13z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
